### PR TITLE
Conditional Template

### LINKED ISSUE #2741: Enable optional fields to show/hide in UI
As templates become more advanced, we need to support a more dynamic form template, implementing support for `allOf`, `if/else` in the template json. 

Some UI-specific changes will need to be made as the jrsf library is pretty fussy about rendering when there are null fields in the template.

### PR DESCRIPTION
**Pull request by damoodamoo**
# Resolves #2741 

- Implements the `allOf` feature of JSON schema to allow conditional fields + conditional schema validation
- "auto_create" string has gone in favour of an enum (dropdown), which shows appropriate follow-up fields
- all workspace templates updated to use conditional schema to streamline the form
- Airlock review VM configuration - nested properties, conditionally required
- UI code to massage the data payload sent in for create or update of a resource. 

image

### CODE REVIEW

File: ui/app/src/models/resourceTemplate.ts
Comment (tanya-borisova) [inline ui/app/src/models/resourceTemplate.ts:22]: FWIW this doesn't look that horrible to me although it is obviously a hack. Perhaps worth a second pair of eyes though

**Review by tanya-borisova — APPROVED** [review]
LGTM in the API app domain but I think it would need another approval for the UI part.

Exciting changes! 🎉

File: api_app/tests_ma/test_api/test_routes/test_airlock.py
Comment (damoodamoo) [reply api_app/tests_ma/test_api/test_routes/test_airlock.py:107]: not in the end, I was able to keep them nested

File: api_app/schemas/azuread.json
Comment (damoodamoo) [reply api_app/schemas/azuread.json:10]: well not really - BUT - there's no harm leaving it here for now. There's a chance we might revisit all these static templates in time.

File: CHANGELOG.md
Comment (damoodamoo) [reply CHANGELOG.md]: yup, good call - have added

File: api_app/api/routes/airlock.py
Comment (damoodamoo) [reply api_app/api/routes/airlock.py]: good spot!

File: ui/app/src/components/shared/create-update-resource/ResourceForm.tsx
Comment (damoodamoo) [reply ui/app/src/components/shared/create-update-resource/ResourceForm.tsx:38]: good spot, i'd reworked that and left it there!

File: ui/app/src/components/shared/create-update-resource/ResourceForm.tsx
Comment (damoodamoo) [reply ui/app/src/components/shared/create-update-resource/ResourceForm.tsx]: faaaaaaaaaaanx!

File: ui/app/src/components/shared/create-update-resource/ResourceForm.tsx
Comment (damoodamoo) [reply ui/app/src/components/shared/create-update-resource/ResourceForm.tsx:34]: yup, broken it up..

**Review by jjgriff93 — APPROVED** [review]
Perfect, looks great!

**Pull request merged**
